Support library name postfix

yukoba opened the pull request:
Recently, some JavaCPP presets library names have postfix.
For example, ```opencv-gpu```, ```mkl-redist```, and ```tensorflow-python-gpu```.
```-gpu```, ```-redist```, and ```-python-gpu``` is postfix.
These postfixes go to ```classifier```.
I fixed this.

Sorry for sending this pull request after @lloydmeta released 1.15.
I just noticed now.

yukoba committed:
Support library name postfix

lloydmeta reviewed (commented):
Can you please add a test for this ?

yukoba commented:
> Can you please add a test for this ?

Can I replace VectorTest to MklTest?
https://github.com/bytedeco/sbt-javacpp/pull/23/commits/765b73476d65d71b30150cc629bdee4ef1c1e31b#diff-44a7c18d14107a86b0338919226e9a05
VectorTest and MklTest cannot coexist.
I think the reason is VectorTest uses Builder and Loader.load().

lloydmeta commented:
@yukoba `VectorTest` is a pretty thorough test and is ported (at the time) from something in JavaCpp itself. It's also a good baseline test so I think we should keep it.

Why does `Builder` and `Loader.load()` interfere with the test? Is there a known issue? Just curious because I don't actually use JavaCpp much.

yukoba commented:
To avoid the problem I forked each test.
I restored VectorTest.

> Why does Builder and Loader.load() interfere with the test? Is there a known issue? Just curious because I don't actually use JavaCpp much.

I don't know too.

saudet commented:
It shouldn't cause any issues. What's the error that you get?

yukoba commented:
@saudet 

> What's the error that you get?

The error message is this.

```
[error] Test javacpp.MklTest.testGetMaxCpuFrequency failed: java.lang.UnsatisfiedLinkError: org.bytedeco.mkl.global.mkl_rt.MKL_CBWR_UNSET_ALL()I, took 0.381 sec
[error]     at org.bytedeco.mkl.global.mkl_rt.MKL_CBWR_UNSET_ALL(Native Method)
[error]     at org.bytedeco.mkl.global.mkl_rt.<clinit>(mkl_rt.java:225)
[error]     at javacpp.MklTest.testGetMaxCpuFrequency(MklTest.java:9)
[error]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error]     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error]     at java.lang.reflect.Method.invoke(Method.java:566)
[error]     ...
```

The complete output
without fork: https://gist.github.com/yukoba/9cfb1399e264e48464fcf2f4d4a828f9
with fork: https://gist.github.com/yukoba/0ea5b98400b55901d4fbd81aa379cf59

My PC is Windows 10 and has MKL 2019.4 and Visual Studio 2019.

yukoba commented:
@saudet 

> My PC is Windows 10 and has MKL 2019.4 and Visual Studio 2019.

The same thing happens on clean Ubuntu 20.04 too.
You can test by this.

```
echo "deb https://dl.bintray.com/sbt/debian /" | sudo tee -a /etc/apt/sources.list.d/sbt.list
curl -sL "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x2EE0EA64E40A89B84B2DF73499E82A75642AC823" | sudo apt-key add
sudo apt update
sudo apt install sbt openjdk-11-jdk g++

git clone -b LibraryNamePostfix https://github.com/yukoba/sbt-javacpp.git
cd sbt-javacpp
sbt compile publishLocal
cd src/sbt-test/sbt-javacpp/simple
sbt -Dplugin.version=1.16-SNAPSHOT test
```

saudet commented:
I'm not able to reproduce this issue with Maven, so it looks like this has something to do with sbt... We may be able to understand better what's happening by looking at the debug messages, for example, by setting the "org.bytedeco.javacpp.logger.debug" system property to "true".

yukoba commented:
> I'm not able to reproduce this issue with Maven, so it looks like this has something to do with sbt

I found one thing.

I need this line,
```Test / fork := true```
but I don't need this line.
```Test / testGrouping := (Test / definedTests).value.map(test => Group(test.name, Seq(test), SubProcess(ForkOptions())))```

I added the second line to split MklTest process and VectorTest process, but this was meaningless.
What I have to do is separate the Sbt process and test process.

I checked both Ubuntu 20.04 and Windows 10.

Sbt running without fork has many limitations and I always fork.
According to this page https://www.scala-sbt.org/1.x/docs/Running-Project-Code.html

> the wrong class loader might be used for various complex reasons

Maybe related to this.

I removed the second line from the pull request.
Also I wrote this to README.markdown.

lloydmeta commented:
Thanks for this, just pushed a new 1.16 release with this changeset.

yukoba commented:
Thank you.
